src/validateInvocation.ts

```diff
@@ -3,20 +3,25 @@ import {
   IntegrationValidationError,
 } from '@jupiterone/integration-sdk-core';
 
+import { calculateConfig } from './calculateConfig';
 import { createAPIClient } from './client';
-import { IntegrationConfig } from './types';
+import { CalculatedIntegrationConfig } from './types';
 
 export default async function validateInvocation(
-  context: IntegrationExecutionContext<IntegrationConfig>,
+  context: IntegrationExecutionContext<CalculatedIntegrationConfig>,
 ) {
   const { config } = context.instance;
 
-  if (!config.clientId || !config.clientSecret) {
+  if (!config.accountId || !config.clientId || !config.clientSecret) {
     throw new IntegrationValidationError(
-      'Config requires all of {clientId, clientSecret}',
+      'Config requires all of {clientId, clientSecret, accountId}',
     );
   }
 
-  const apiClient = createAPIClient(config);
+  const calculatedConfig = calculateConfig(context);
+
+  const apiClient = createAPIClient(calculatedConfig);
   await apiClient.verifyAuthentication();
+
+  context.instance.config = calculatedConfig;
```

Comment:
Understood! It makes sense because like you said, only a few requests are using those calculated values. We leaned heavily towards Qualys (first time making use of executionHistory) so that's why it was like that (which you've guessed at later message!).

I've combined a few of your messages on this topic into something that works as you asked for - but please do take another look at it and let us know if it needs some additional changes.